test/validators/register.ts

```diff
@@ -1,53 +1,133 @@
+    });
+
+    it('Register validator emits ValidatorAdded event', async () => {
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.publicKey(1),
+            helpers.DataGenerator.pod.new(),
+            helpers.DataGenerator.shares(0),
+            '10000'
+        )).to.emit(ssvNetworkContract, 'ValidatorAdded');
+    });
+
+    it('Register one validator into an empty pod', async () => {
+        await helpers.registerValidators(0, 1, '10000', helpers.DataGenerator.pod.new(), [GasGroup.REGISTER_VALIDATOR_NEW_STATE]);
+    });
+
+    it('Register two validators into an existing pod', async () => {
+        await helpers.registerValidators(0, 1, '10000', helpers.DataGenerator.pod.byId(podResult.podId), [GasGroup.REGISTER_VALIDATOR_EXISTED_POD]);
+    });
+
+    it('Register two validators into an existing cluster', async () => {
+        await helpers.registerValidators(1, 1, '10000', helpers.DataGenerator.pod.byId(podResult.podId), [GasGroup.REGISTER_VALIDATOR_EXISTED_CLUSTER]);
+    });
+
+    it('Invalid operator amount', async () => {
+        // 2 Operators
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2],
+            helpers.DataGenerator.shares(0),
+            '10000'
+        )).to.be.revertedWith('OessDataStructureInvalid');
+
+        // 6 Operators
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2, 3, 4, 5, 6],
+            helpers.DataGenerator.shares(0),
+            '10000'
+        )).to.be.revertedWith('OessDataStructureInvalid');
+    });
+
+    it('Invalid public key length', async () => {
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.shares(0),
+            [1, 2, 3, 4],
+            helpers.DataGenerator.shares(0),
+            '10000'
+        )).to.be.revertedWith('InvalidPublicKeyLength');
+    });
+
+    it('Not enough amount', async () => {
+        await expect(ssvNetworkContract.connect(helpers.DB.owners[1]).registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2, 3, 4],
+            helpers.DataGenerator.shares(0),
+            '1'
+        )).to.be.revertedWith('account liquidatable');
+    });
+
+    // GAS AMOUNT IS ABOVE EXPECTED
+    it('Register with 7 operators', async () => {
+        await helpers.registerValidators(0, 1, '10000', helpers.DataGenerator.pod.new(7), [GasGroup.REGISTER_VALIDATOR_EXISTED_POD]);
+    });
+
+    // THIS NEEDS VALIDITY (I ADDED IN SOLIDITY CODE)
+    it('Non existent operator', async () => {
+        await expect(ssvNetworkContract.registerValidator(
+            helpers.DataGenerator.publicKey(0),
+            [1, 2, 3, 25],
+            helpers.DataGenerator.shares(0),
+            '4000'
+        )).to.be.revertedWith('OperatorDoesntExist');
+    });
```

Comment:
I already added validity in solidity but note that non existent operator needs validity (in case my solidity code is rejected)